pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -727,19 +727,23 @@ def _handle_distribution_tree(declarative_content):
             if variants:
                 Variant.objects.bulk_create(variants)
 
-        update_references_to_save = []
-        update_collection_packages_to_save = []
-        modulemd_pkgs = []
-        group_pkgs = []
-        category_groups = []
-        env_groups = []
-        env_optgroups = []
+        UpdateRecordCollections = UpdateRecord.collections.through
         ModulemdPackages = Modulemd.packages.through
         PackageGroup_packages = PackageGroup.related_packages.through
         PackageCategory_groups = PackageCategory.packagegroups.through
         PackageEnvironment_groups = PackageEnvironment.packagegroups.through
         PackageEnvironment_optionalgroups = PackageEnvironment.optionalgroups.through
 
+        update_collection_to_save = MaxSizeBulk(UpdateCollection, 100)
```

Comment:
```bulk_create()``` has a batch size argument, is there a reason we cannot use that? https://docs.djangoproject.com/en/dev/ref/models/querysets/#bulk-create

Reply:
Yesterday @dkliban said he was suspicious that memory was crashing before hitting the saving point, so I built this object to bulk create as soon it reaches the `max_length`, so it does not need to wait till the end of the for loop.